packages/shorebird_cli/lib/src/artifact_manager.dart

```diff
@@ -377,6 +409,26 @@ class ArtifactManager {
     return releaseSupplementDir;
   }
 
+  /// Returns the path to the shorebird release supplement directory for macOS.
+  ///
+  /// Returns null if there is no supplement directory
+  /// (e.g. when using older Flutter revisions).
+  Directory? getMacosReleaseSupplementDirectory() {
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    final releaseSupplementDir = Directory(
+      p.join(projectRoot.path, 'build', 'macos', 'shorebird'),
+    );
+
+    if (!releaseSupplementDir.existsSync()) {
```

Comment:
This should technically never happen so we might want to have a more fatal error here since MacOS was only introduced after the supplement artifact was introduced.